assembly/sdk/collections/persistentUnorderedMap.ts

```diff
@@ -196,23 +196,28 @@ export class PersistentUnorderedMap<K, V> {
    * @param key Key to remove.
    */
   delete(key: K): void {
-    // make sure the item is in the set
```

Comment:
faster delete when we're deleting the last (lenght-1) key
also get index && check key exists in the same operation